bpf/ringbuf.h

```diff
@@ -21,7 +21,7 @@
 // we can share them later if we find is worth not including code per duplicate
 struct {
     __uint(type, BPF_MAP_TYPE_RINGBUF);
-    __uint(max_entries, 1 << 24);
+    __uint(max_entries, 1 << 16);
```

Comment:
As max_entries are allocated dynamically as long as more space is needed, this should not affect in terms of memory (unless we end up having more than 2^16 events in the queue).

Reply:
When I printed BPF metrics, I see the whole ring buffer pre-allocated in locked memory.

Before:
```
870: ringbuf  name events  flags 0x0
	key 0B  value 0B  max_entries 16777216  memlock 16,855,384B
```

After:
```
1079: ringbuf  name events  flags 0x0
	key 0B  value 0B  max_entries 65536  memlock 78,424B
```

There are more issues with what we do. One example is that we'll load two separate programs when we have `gin`. Once for the `gin` functions, but we'll load the go http too, because it's there in the binary. So in a sense, we'll duplicate the memory and have 2 ring buffers when we instrument with `gin`.

Another issue is grpc, `golang.org/x/net/http2/hpack.(*Encoder).WriteField` belongs to the gosdk, so we load both http and grpc for grpc only applications, again two ring buffers, all maps are duplicate :(.